examples/src/examples/misc/hello-script-modules.mjs

```diff
@@ -0,0 +1,95 @@
+import * as pc from 'playcanvas';
+
+/**
+ * @param {import('../../options.mjs').ExampleOptions} options - The example options.
+ * @returns {Promise<pc.AppBase>} The example application.
+ */
+async function example({ canvas, deviceType, glslangPath, twgslPath }) {
+
+    const gfxOptions = {
+        deviceTypes: [deviceType],
+        glslangUrl: glslangPath + 'glslang.js',
+        twgslUrl: twgslPath + 'twgsl.js'
+    };
+
+    const device = await pc.createGraphicsDevice(canvas, gfxOptions);
+    const createOptions = new pc.AppOptions();
+    createOptions.mouse = new pc.Mouse(document.body);
+    createOptions.graphicsDevice = device;
+
+    createOptions.componentSystems = [
+        // @ts-ignore
+        pc.EsmScriptComponentSystem,
+        // @ts-ignore
+        pc.RenderComponentSystem,
+        // @ts-ignore
+        pc.CameraComponentSystem,
+        // @ts-ignore
+        pc.LightComponentSystem
+
+    ];
+    createOptions.resourceHandlers = [
+        // @ts-ignore
+        pc.TextureHandler,
+        // @ts-ignore
+        pc.ContainerHandler
+    ];
+
+    const app = new pc.AppBase(canvas);
+    app.init(createOptions);
+    app.start();
+
+    // Set the canvas to fill the window and automatically change resolution to be the same as the canvas size
+    app.setCanvasFillMode(pc.FILLMODE_FILL_WINDOW);
+    app.setCanvasResolution(pc.RESOLUTION_AUTO);
+
+    // Ensure canvas is resized when window changes size
+    const resize = () => app.resizeCanvas();
+    window.addEventListener('resize', resize);
+    app.on('destroy', () => {
+        window.removeEventListener('resize', resize);
+    });
+
+    // create box entity
+    const box = new pc.Entity('World');
+    box.addComponent('render', {
+        type: 'box'
+    });
+
+    box.addComponent('esmscript', {
+        enabled: true,
+        modules: [
+            {
+                moduleSpecifier: '/scripts/make-confetti.js',
```

Comment:
It is also possible to have this esmscript class inside the same file? 
(it's not a requirement, just curious)

Reply:
Potentially. Currently there the api is `entity.esmscript.import('./path/to/esm-script.js')` which imports the module, but also uses the path to listen for editor updates. If hot editor hot swapping isn't needed, which is all non-editor projects, then I was considering a `entity.esmscript.create(Module)` which just does the same but without the ability to hot-swap.

Reply:
Can hot-swapping be made available in non-editor projects? I'm running a game using modules without bundling .. and when modify a module, can it reload using hot-swapping?

Reply:
Hot swapping is only available with [webpack](https://webpack.js.org/concepts/hot-module-replacement/) or Vite etc. It's possible to get that working, but requires separate api